flow/project.py

```diff
@@ -1193,6 +1194,8 @@ def __new__(metacls, name, bases, namespace):
         cls._OPERATION_FUNCTIONS = []
         cls._OPERATION_PRECONDITIONS = defaultdict(list)
         cls._OPERATION_POSTCONDITIONS = defaultdict(list)
+        cls._OPERATION_HOOK_REGISTRY = defaultdict(lambda: defaultdict(list))
+        cls._operation_hooks = defaultdict(list)
```

Comment:
Why is this defined here? This is initialized in `__init__` for the `FlowProject`.
```suggestion
```